### ISSUE TITLE
<head> element incomplete when generated via EpubHtml.get_content()

### ISSUE THREAD

**Issue by andyroberts** [aerkalov/ebooklib#76]
I should declare upfront that I've encountered this issue using v0.15 installed via pip, but looking at master branch it looks like the issue remains. 

Perhaps this is a deliberate design choice -- in which case it ought to be documented -- but the <head> element will only contain CSS or Javascript items as found in the EpubHtml.items list. Otherwise EpubHtml.get_content() will return HTML with an empty head element.

The code that once did populate the head element has been commented out but still resides within the code (https://github.com/aerkalov/ebooklib/blob/master/ebooklib/epub.py#L265-L271):

```python
      # this should not be like this
      # head = html_root.find('head')
      # if head is not None:
      #     for i in head.getchildren():
      #         if i.tag == 'title' and self.title != '':
      #             continue
      #         _head.append(i)
```

As it stands, if one uses EbookLib to load an epub, and then write it back out without any other manipulation then all elements within the original head tag are omitted and the resulting file has an empty <head>, although I suppose one could argue that's a different bug altogether.

**Comment by aerkalov** [comment]
Correct, this is a bug. As I recall it correctly the issue was (in my project) that some things would end up multiple times in <head>. They were parsed and they were additionally added by the system. Some validation when additional resources or metadata is added to the item should be implemented instead of this. I never really used it for some real EPUB=>EPUB conversion so it stayed unnoticed.

**Comment by andyroberts** [comment]
@aerkalov Is this lib no longer being maintained?

**Comment by aerkalov** [comment]
Hey Andy! No, it is being maintained and I use it in pretty big project myself. Will check these things over the weekend, my mind is at the moment occupied with some other things.

**Comment by andyroberts** [comment]
Great. I'm only checking so that I know it's worth opening pull requests.

**Comment by danielhjames** [comment]
Hi @aerkalov Would this be good to merge now? Just checking in case I need to update https://packages.debian.org/sid/python-ebooklib etc.

**Comment by tadeoos** [comment]
I run into this issue and solved it (for CSS files, easy to generalize) by making sure the `title` and `.links` attribute is populated correctly on page load via this plugin:

```python

def strip_leading_parent_dirs(path: str) -> str:
    return re.sub(r"^(?:\.\./)+", "", path)

class LoadMissingHeaderItems(BasePlugin):
    @classmethod
    def html_after_read(cls, book: epub.EpubBook, item: epub.EpubHtml):
        soup = BeautifulSoup(item.content, "html.parser")
        try:
            item.title = soup.head.title.get_text()
        except AttributeError:
            pass
        for el in soup.head.find_all("link"):
            linked_item = book.get_item_with_href(strip_leading_parent_dirs(el["href"]))
            if linked_item:
                item.add_item(linked_item)
            else:
                raise RuntimeError(f"Has a linked item which wasn't found: {el}")
```

Sharing in case someone needs an inspiration for a workaround, but it would be good to solve this in the library itself.
